regression/pages/common/api_clients.py

```diff
@@ -0,0 +1,260 @@
+"""
+API classes for different tasks
+"""
+import Cookie
+import json
+
+import requests
+from requests.auth import AuthBase
+from edx_rest_api_client.client import EdxRestApiClient
+
+from regression.pages.whitelabel.const import (
+    ACCESS_TOKEN,
+    AUTH_PASSWORD,
+    AUTH_USERNAME,
+    ECOMMERCE_API_URL,
+    ENROLLMENT_API_URL,
+    URL_WITHOUT_AUTH
+)
+
+
+class ApiException(Exception):
+    """
+    Exceptions for API failures
+    """
+    pass
+
+
+def check_response(response):
+    """
+    Check whether a response was successful. If not, raise an exception
+    Args:
+        response:
+    """
+    if response.status_code != 200:
+        raise ApiException(
+            'API request failed with following error code: ' +
+            str(response.status_code)
+        )
+
+
+class BearerAuth(AuthBase):
+    """
+    Attaches Bearer Authentication to the given Request object.
+    """
+
+    def __init__(self, token):
+        """ Instantiate the auth class. """
+        self.token = token
+
+    def __call__(self, r):
+        """ Update the request headers. """
+        r.headers['Authorization'] = 'Bearer {}'.format(self.token)
+        return r
+
+
+class EcommerceApiClient(object):
+    """
+    Ecommerce API client for various actions like creating, deleting
+    downloading coupons
+    """
+
+    @property
+    def ecommerce_api_client(self):
+        """
+        E-Commerce API Client
+        """
+        return EdxRestApiClient(
+            ECOMMERCE_API_URL, oauth_access_token=ACCESS_TOKEN)
+
+    def create_coupon(self, payload):
+        """
+        Create coupons using API
+        Args:
+            payload:
+        """
+        response = self.ecommerce_api_client.coupons.post(data=payload)
+        if 'coupon_id' not in response.keys():
+            raise ApiException('Coupon not created')
+        return response['coupon_id']
+
+    def get_coupon_report(self, coupon_id):
+        """
+        Get coupon report using api
+        Args:
+            coupon_id:
+        """
+        response = self.ecommerce_api_client.products(coupon_id).get()
+        if not response:
```

Comment:
Client only returns a dictionary of values and does not return any status code or something so I could not check the 200 or ok status. In case of any errors the client will raise a slumber http exception which i don't think we need to catch, the only eventuality I am covering here is that if client returns an empty list or dictionary. sometimes the coupon ids can exist without any valid data attached with these so in that scenario I would like to raise an exception